engine/src/main/java/com/datatorrent/stram/client/StramAppLauncher.java

```diff
@@ -543,8 +543,9 @@ public URLClassLoader loadDependencies()
 
   private void setTokenRefreshCredentials(LogicalPlan dag, Configuration conf) throws IOException
   {
-    String principal = StramUserLogin.getPrincipal();
-    String keytabPath = conf.get(StramClientUtils.KEY_TAB_FILE);
+    String principal = conf.get(StramClientUtils.TOKEN_REFRESH_PRINCIPAL, StramUserLogin.getPrincipal());
+    String keytabPath = principal.equals(StramUserLogin.getPrincipal()) ? conf.get(StramClientUtils.KEY_TAB_FILE, null) : conf.get(StramClientUtils.TOKEN_REFRESH_KEYTAB, null);
```

Comment:
Looks like TOKEN_REFRESH_KEYTAB is just a new replacement name for the old KEY_TAB_FILE property as it is complementary to TOKEN_REFRESH_PRINCIPAL. Shouldn't it just be 
String keytabPath = conf.get(StramClientUtils.TOKEN_REFRESH_KEYTAB, StramClientUtils.KEY_TAB_FILE);